### PR TITLE
feat(admin): add custom admin widget ranking feature

### LINKED ISSUE #15240: Admin: allow ordering custom widgets within an injection zone
### Problem

Custom admin widgets injected into the same zone (e.g. `product.details.after`) currently appear in an arbitrary order based on file discovery. With multiple plugins or several local widgets sharing a zone, there is no way for the developer to control which widget appears first.

### Use cases

- A plugin author shipping multiple widgets for the same zone wants them displayed in a deterministic, designed order.
- A team using several plugins wants to position a specific plugin's widget above another's, without forking either plugin.
- An app developer combining their own widgets with plugin widgets wants their own to render first.

### Expected behavior

The `WidgetConfig` exposed by `@medusajs/admin-sdk` should accept an optional `rank: number` property. Widgets within the same injection zone are then sorted ascending by rank (lower rank first), and unranked widgets appear after all ranked ones, preserving their relative discovery order. Sorting must happen after all plugins have contributed their widgets to the zone, so a `rank: 1` widget from plugin B appears before a `rank: 5` widget from plugin A.

### Prior art

This mirrors the ranking system already in place for [UI routes](https://github.com/medusajs/medusa/pull/13946) and aligns with the menu item ranking handled by `populateMenus`.

### Proposed PR

#14424.

### PR DESCRIPTION
**Pull request by bqst**
Closes #15240

### What — What changes are introduced in this PR?

This PR adds a `rank` property to the `WidgetConfig` interface, allowing developers to control the order of custom admin widgets within the same injection zone. Widgets are sorted in ascending order by rank value (lower rank appears first), with unranked widgets appearing after all ranked widgets.

### Why — Why are these changes relevant or necessary?

When developers create multiple custom widgets for the same injection zone (e.g., `product.details.after`), they need a way to control the display order. Without ranking, widgets appear in an arbitrary order based on file discovery, making it difficult to create a predictable UI. This feature follows the same pattern already implemented for [UI routes ranking](https://github.com/medusajs/medusa/pull/13946), providing consistency across the admin extension system.

### How — How have these changes been implemented?

1. **Type Definitions** - Added `rank?: number` property to `WidgetConfig` interface in `@medusajs/admin-sdk` and `WidgetExtension` type in `@medusajs/dashboard`
2. **Config Parsing** - Updated the vite plugin's `generate-widgets.ts` to extract and include rank values from widget configs, including support for negative numbers (parsed as `UnaryExpression` in Babel AST)
3. **Sorting Logic** - Implemented `sortWidgetsByRank()` utility in the dashboard app that:
   - Sorts widgets by rank in ascending order
   - Places widgets with rank before widgets without rank
   - Maintains original order for widgets with equal or undefined ranks
4. **Babel Helpers** - Added `isUnaryExpression` export to handle negative rank values
5. **Tests** - Added comprehensive tests for widget generation and sorting logic

## Testing — How have these changes been tested, or how can the reviewer test the feature?

✅ **Automated tests added** - All passing
- 4 tests for widget generation (including negative rank)
- 8 tests for sorting logic (including edge cases)

✅ **Manual testing** with multiple widgets in the same zone demonstrating:
- Positive rank values
- Negative rank values
- Mixed ranked and unranked widgets

### Basic Usage

```tsx
// src/admin/widgets/product-widget-first.tsx
import { defineWidgetConfig } from "@medusajs/admin-sdk"

const ProductWidgetFirst = () => {
  return <div>First Widget</div>
}

export const config = defineWidgetConfig({
  zone: "product.details.after",
  rank: 1, // Will appear first
})

export default ProductWidgetFirst
```

```tsx
// src/admin/widgets/product-widget-second.tsx
import { defineWidgetConfig } from "@medusajs/admin-sdk"

const ProductWidgetSecond = () => {
  return <div>Second Widget</div>
}

export const config = defineWidgetConfig({
  zone: "product.details.after",
  rank: 2, // Will appear second
})

export default ProductWidgetSecond
```

### Negative Ranks (appear before positive ranks)

```tsx
export const config = defineWidgetConfig({
  zone: "product.details.after",
  rank: -1, // Will appear before rank: 1
})
```

### Mixed Ranked and Unranked Widgets

```tsx
// Widgets without rank appear after all ranked widgets
export const config = defineWidgetConfig({
  zone: "product.details.after",
  // No rank - will appear last
})
```

## Checklist

- [x] I have added a changeset for this PR
- [x] The changes are covered by relevant tests
- [x] I have verified the code works as intended locally

### Implementation Notes

- **Pattern Consistency**: Follows the same rank pattern as UI routes ranking (#13946)
- **Backward Compatible**: Widgets without rank continue to work (appear after ranked widgets)
- **Zero and Negative Ranks**: Fully supported
- **Stable Sort**: Widgets with equal ranks maintain their original order

### Files Changed

- `packages/admin/admin-sdk/src/config/types.ts` - Added `rank` to `WidgetConfig`
- `packages/admin/admin-vite-plugin/src/babel.ts` - Added `isUnaryExpression` export
- `packages/admin/admin-vite-plugin/src/widgets/generate-widgets.ts` - Extract rank from config
- `packages/admin/dashboard/src/dashboard-app/types.ts` - Added `rank` to `WidgetExtension`
- `packages/admin/dashboard/src/dashboard-app/dashboard-app.tsx` - Sort widgets by rank
- `packages/admin/dashboard/src/dashboard-app/utils/sort-widgets-by-rank.ts` - New sorting utility

---

> [!NOTE]
> Introduces deterministic ordering for admin widgets via an optional `rank`.
> 
> - Adds `rank?: number` to `WidgetConfig` (`@medusajs/admin-sdk`) and `WidgetExtension` (`@medusajs/dashboard`)
> - Updates `admin-vite-plugin` to parse and emit `rank` from widget configs, including negative values (adds `isUnaryExpression` in `babel.ts`); codegen now includes `rank` in `widgets` output
> - Dashboard now sorts widgets per zone using `sortWidgetsByRank` and applies it in `dashboard-app.tsx`
> - Adds tests for widget generation (incl. negative ranks) and sorting utility
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 85e86c80dd8b0c69899223107efb8fc9d7e76379. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by docloulou** [comment]
bump

**Comment by bqst** [comment]
unstale

Linked issue opened (#15240) and added `Closes #15240` to the description.

On the `populateWidgets` concern from @medusa-os-bot: the implementation already follows the same pattern as `populateMenus`. In `dashboard-app.tsx` (`populateWidgets`), the outer `plugins.forEach` collects every widget into a per-zone `tempRegistry`. Sorting only happens afterwards, in the second `tempRegistry.forEach` block, so a `rank: 1` widget from plugin B will end up before any unranked or higher-ranked widgets from plugin A. There is no per-plugin sort step. Tests cover the multi-plugin case as well.

Also merged latest `develop` into the branch.

File: packages/admin/admin-vite-plugin/src/widgets/generate-widgets.ts [hunk 1]
```diff
@@ -108,13 +115,12 @@ async function parseFile(
     return null
   }
 
-  if (!zone) {
-    logger.warn(`'zone' property is missing from the widget config.`, { file })
+  if (!widgetConfig) {
     return null
   }
```
Comment (shahednasser) [inline packages/admin/admin-vite-plugin/src/widgets/generate-widgets.ts:120]: The original warning was useful to the users, we should keep it.

**Review by shahednasser — CHANGES_REQUESTED** [review]
Small change, also please address the OS Bot's comment
